proposals/lip_define_state_and_state_transitions_of_dpos_module.md

```diff
@@ -0,0 +1,1160 @@
+        }
+    }
+}
+```
+
+
+##### Verification
+
+Both properties of the parameters must follow the block header schema as defined in [LIP "New block header and block asset schema"][LIP-newBlockHeader]. 
+Validity of this transaction is then specified in [LIP 0024][LIP-0024-verifyPOM].
+
+
+##### Execution
+
+Execution of this transaction is specified in [LIP 0024][LIP-0024-applyingPOM].
+
+
+### Internal Functions
+
+
+#### Round Number and End of Rounds
+
+All blocks (with the exception of the genesis block) are part of a round. 
+The first block after the genesis block is the first block of the first round, and so on. 
+The round length, i.e. the number of blocks in a round, is specified in a configuration file and is denoted `ROUND_LENGTH`. 
+In the block lifecycle, it will be useful to compute the round number to which a block belongs and if the block is the last block of its round. 
+For this, we will use the following functions:
+
+
+##### roundNumber
+
+This function returns the round number to which the input height belongs.
+
+```python
+roundNumber(h):
+    return ceiling((h - genesisHeight) / ROUND_LENGTH)
+```
+
+
+##### isEndOfRound
+
+This function returns a boolean indicating if the input height is at the end of a round or not.
+
+
+```python
+isEndOfRound(h):
+    if (h - genesisHeight) % ROUND_LENGTH == 0:
+        return True
+    else:
+        return False
+```
+
+
+#### delegateWeight
+
+The delegate weight is always a function of the votes, the potential misbehaviors and the block height.
+
+
+##### Parameters
+
+The function has the following input parameters in the order given below:
+
+* `address`: A 20-byte addresses of a delegate.
+* `height`: The height for which the weight is computed.
+
+
+##### Returns
+
+This function returns the delegate weight.
+
+
+##### Execution
+
+```python
+delegateWeight(address, height):
+
+    if there exist h in delegateStore(address).pomHeights with 0 < height - h < PUNISHMENT_WINDOW:
+        return 0
+    else:
+        return min(delegateStore(address).selfVotes * FACTOR_SELF_VOTES,
+                   delegateStore(address).totalVotesReceived)
+```
+
+
+
+#### shuffleValidatorsList
+
+A function to reorder the list of validators as specified in [LIP 0003][LIP-0003].
+
+
+##### Parameters
+
+The function has the following input parameters in the order given below:
+
+* `validatorsAddresses`: An array of pairwise distinct 20-byte addresses.
+* `randomSeed`: A 32-byte value representing a random seed.
+
+
+##### Returns
+
+This function returns an array of bytes with the re-ordered list of addresses.
+
+
+##### Execution
+
+```python
+shuffleValidatorsList(validatorsAddresses, randomSeed):
+    
+    roundHash = {}
+    for address in validatorsAddresses:
+        roundHash[address] = hash(randomSeed || address)
+
+    # Reorder the validator list
+    shuffledValidatorAddresses = sort validatorsAddresses where address1 < adress2 if (roundHash(address1) < roundHash(address2))
```

Comment:
```suggestion
    shuffledValidatorAddresses = sort validatorsAddresses where address1 < address2 if (roundHash(address1) < roundHash(address2))
```

Reply:
@IkerAlus you might want to fix this too

Reply:
fixed